### DISCUSSION TITLE
Upgrade 3.4.5 to 3.4.8 error

### DISCUSSION THREAD

**Discussion by darrenwoods3** [netbox-community/netbox#12313] [Help Wanted!] [answered]
I'm receiving this error when trying to upgrade.  Any help will be appreciated!!

@netbox-01:~$ cd /opt/netbox/

@netbox-01:/opt/netbox$ sudo git checkout master
M       contrib/apache.conf
Already on 'master'
Your branch is behind 'origin/master' by 120 commits, and can be fast-forwarded.
  (use "git pull" to update your local branch)

@netbox-01:/opt/netbox$ sudo git pull origin master
From https://github.com/netbox-community/netbox
 * branch                master     -> FETCH_HEAD
Updating 6638fd88b..3c91331e1
error: Your local changes to the following files would be overwritten by merge:
        contrib/apache.conf
Please commit your changes or stash them before you merge.
Aborting

**Comment by mtinberg** [comment, marked as answer]
The problem is as it says in the error message, the example contrib/apache.conf file has been modified so git will refuse to lose or overwrite your changes.  git diff contrib/apache.conf to see what has changed, if this change is intentional and this file is being used, you'll want to copy this file to a more appropriate location as this isn't something that expected to be modified in-place as part of a normal install.  When it is safe to do so you can git checkout contrib/apache.conf to replace the file with the version from the git database, then your git pull should be able to run without (the same) error.


—

Mark Tinberg ***@***.***>

Division of Information Technology-Network Services

University of Wisconsin-Madison


________________________________
From: darrenwoods3 ***@***.***>
Sent: Thursday, April 20, 2023 7:52 PM
To: netbox-community/netbox ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [netbox-community/netbox] Upgrade 3.4.5 to 3.4.8 error (Discussion #12313)


I'm receiving this error when trying to upgrade. Any help will be appreciated!!

@netbox-01:~$ cd /opt/netbox/

@netbox-01:/opt/netbox$ sudo git checkout master
M contrib/apache.conf
Already on 'master'
Your branch is behind 'origin/master' by 120 commits, and can be fast-forwarded.
(use "git pull" to update your local branch)

@netbox-01:/opt/netbox$ sudo git pull origin master
From https://github.com/netbox-community/netbox

  *   branch master -> FETCH_HEAD
Updating 6638fd8<https://github.com/netbox-community/netbox/commit/6638fd88b4345647232aeb0f0500ec2c6e577867>..3c91331<https://github.com/netbox-community/netbox/commit/3c91331e167242f2879a5f020016cb3cb7c90cb8>
error: Your local changes to the following files would be overwritten by merge:
contrib/apache.conf
Please commit your changes or stash them before you merge.
Aborting

**Comment by darrenwoods3** [comment]
Mark, thanks for the information.  This resolved my upgrade issue.

Darren Woods

From: Mark Tinberg ***@***.***>
Date: Thursday, April 20, 2023 at 8:17 PM
To: netbox-community/netbox ***@***.***>
Cc: Woods, Darren L ***@***.***>, Author ***@***.***>
Subject: Re: [netbox-community/netbox] Upgrade 3.4.5 to 3.4.8 error (Discussion #12313)

**THIS EMAIL IS FROM AN EXTERNAL SENDER OUTSIDE OF THE ALAMO COLLEGES.
Be cautious before clicking links or opening attachments from unknown sources. Do not provide personal or confidential information.**

The problem is as it says in the error message, the example contrib/apache.conf file has been modified so git will refuse to lose or overwrite your changes. git diff contrib/apache.conf to see what has changed, if this change is intentional and this file is being used, you'll want to copy this file to a more appropriate location as this isn't something that expected to be modified in-place as part of a normal install. When it is safe to do so you can git checkout contrib/apache.conf to replace the file with the version from the git database, then your git pull should be able to run without (the same) error.


—

Mark Tinberg ***@***.***>

Division of Information Technology-Network Services

University of Wisconsin-Madison


________________________________
From: darrenwoods3 ***@***.***>
Sent: Thursday, April 20, 2023 7:52 PM
To: netbox-community/netbox ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [netbox-community/netbox] Upgrade 3.4.5 to 3.4.8 error (Discussion #12313)


I'm receiving this error when trying to upgrade. Any help will be appreciated!!

@netbox-01:~$ cd /opt/netbox/

@netbox-01:/opt/netbox$ sudo git checkout master
M contrib/apache.conf
Already on 'master'
Your branch is behind 'origin/master' by 120 commits, and can be fast-forwarded.
(use "git pull" to update your local branch)

@netbox-01:/opt/netbox$ sudo git pull origin master
From https://smex-ctp.trendmicro.com:443/wis/clicktime/v1/query?url=https%3a%2f%2fgithub.com%2fnetbox%2dcommunity%2fnetbox&umid=8d3eda25-c3e4-4912-803c-371b2e3b434c&auth=208ec4c2b43ad41c2409f492bc3b9a6d140ef3bd-d216290ae3fb229f5a541373f6f998e7193e02cc

* branch master -> FETCH_HEAD
Updating 6638fd8<https://github.com/netbox-community/netbox/commit/6638fd88b4345647232aeb0f0500ec2c6e577867>..3c91331<https://github.com/netbox-community/netbox/commit/3c91331e167242f2879a5f020016cb3cb7c90cb8>
error: Your local changes to the following files would be overwritten by merge:
contrib/apache.conf
Please commit your changes or stash them before you merge.
Aborting
